test_runner/regress/test_auth.py

```diff
@@ -82,6 +84,115 @@ def test_compute_auth_to_pageserver(neon_env_builder: NeonEnvBuilder):
             assert cur.fetchone() == (5000050000,)
 
 
+def test_pageserver_multiple_keys(neon_env_builder: NeonEnvBuilder):
+    neon_env_builder.auth_enabled = True
+    env = neon_env_builder.init_start()
+
+    pageserver_token_old = env.auth_keys.generate_pageserver_token()
+    pageserver_http_client_old = env.pageserver.http_client(pageserver_token_old)
+
+    pageserver_http_client_old.reload_auth_validation_keys()
+
+    env.auth_public_key_path = (
+        Path(env.repo_dir) / "auth_public_key.pem" / "auth_public_key_old.pem"
+    )
+    os.rename(
+        Path(env.repo_dir) / "auth_public_key.pem", Path(env.repo_dir) / "auth_public_key.pem.file"
+    )
+    os.mkdir(Path(env.repo_dir) / "auth_public_key.pem")
+    os.rename(
+        Path(env.repo_dir) / "auth_public_key.pem.file",
+        Path(env.repo_dir) / "auth_public_key.pem" / "auth_public_key_old.pem",
+    )
```

Comment:
It took me a long time to understand what this test does. It replaces the file `auth_public_key.pem`, with a _directory_ with the same name `auth_public_key.pem`. That's pretty confusing. Can we change the path in the pageserver config instead? At least add a comment to explain what this does.

Reply:
I originally explored customizing the path, but it is handed through a lot of layers (including the one in the `control_plane` directory) until it finally lands in the pageserver's config. I thought that doing this many changes just to have a nicer directory name isn't that useful.

Good point about it being confusing, I will add a comment and see what other reviewers think about it.

Reply:
Comment added